app/src/shared/components/TextControl/index.jsx

```diff
@@ -45,7 +45,8 @@ const TextControl = ({
     const el = useRef(null)
     const ref = innerRef || el
     const reverted: Ref<boolean> = useRef(false)
-    const [value, setValue] = useState(valueProp || '')
+    const sanitizedValue = valueProp == null ? '' : valueProp
+    const [value, setValue] = useState(sanitizedValue)
```

Comment:
@timoxley, you were right. Same logic does apply to other places (all `setValue(valueProp)`). Good catch!